charts/drax/templates/secret-grafana-slack-contactpoints.yaml

```
@@ -0,0 +1,34 @@
+{{- if $.Values.global.alerting.enabled -}}
+{{- if eq $.Values.global.alerting.slackToken "" -}}
+  {{- fail "Error: global.alerting.slackToken must be defined." -}}
+{{- end -}}
+{{- end -}}
+
+{{- $secretName := printf "%s-slack-auth" $.Release.Name -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ $secretName }}
+  annotations:
+    "helm.sh/resource-policy": keep
+type: Opaque
+data:
+  {{- $existingSecret := lookup "v1" "Secret" .Release.Namespace $secretName }}
+  {{- if $.Values.global.alerting.enabled -}}
+    {{- if and $existingSecret $existingSecret.data (hasKey $existingSecret.data "slack-token") -}}
+      {{- $slackToken := get ($existingSecret.data) "slack-token" | b64dec }}
+      {{- if eq $slackToken "" -}}
+        slack-token: {{ $.Values.global.alerting.slackToken | b64enc | quote }}
+      {{- else }}
+        slack-token: {{ get ($existingSecret.data) "slack-token" | quote }}
```

We've already put this in a variable, so we might as well use it and make it a little bit more readable. I also find it a bit clearer as the base64 encoding is a bit more explicit.

```suggestion
        slack-token: {{ $slackToken | b64enc | quote }}
```